apps/web/app/layout.tsx

```diff
@@ -78,15 +80,12 @@ export const metadata = {
   },
 };
 
-const getInitialProps = async (url: string) => {
-  const { pathname, searchParams } = new URL(url);
-
-  const isEmbed = pathname.endsWith("/embed") || (searchParams?.get("embedType") ?? null) !== null;
-  const embedColorScheme = searchParams?.get("ui.color-scheme");
-
-  const req = { headers: await headers(), cookies: await cookies() };
-  const newLocale = (await getLocale(req)) ?? "en";
-  const direction = dir(newLocale);
+const getInitialProps = async () => {
+  const h = await headers();
+  const isEmbed = h.get("x-isEmbed") === "true";
+  const embedColorScheme = h.get("x-embedColorScheme");
```

Comment:
- these special headers are set in middleware (embed pages go through middleware)